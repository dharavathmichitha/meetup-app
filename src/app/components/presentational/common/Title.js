import React from 'react';
import PropTypes from 'prop-types';

const Title = ({ text, isFree }) => {
  const freeLabel = isFree ? ' FREE!!!' : null;
  return (
    <div className="event__title">
      {text}
      <span className="event__title--free">{freeLabel}</span>
    </div>
  );
};

Title.prototype = {
  text: PropTypes.string.isRequired,
};
export default Title;
