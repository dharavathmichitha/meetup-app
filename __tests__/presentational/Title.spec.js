import React from 'react';
import renderer from 'react-test-renderer';
import Title from '../../src/app/components/presentational/common/Title';

describe('Event Component', () => {
  it('should match snapshot when title does not include free label', () => {
    const props = {
      isFree: false,
      text: 'testText',
    };
    const component = renderer.create(<Title {...props} />);

    const componentTree = component.toJSON();

    expect(componentTree).toMatchSnapshot();
  });
  it('should match snapshot when title includes free label', () => {
    const props = {
      isFree: true,
      text: 'testText',
    };
    const component = renderer.create(<Title {...props} />);

    const componentTree = component.toJSON();

    expect(componentTree).toMatchSnapshot();
  });
});
