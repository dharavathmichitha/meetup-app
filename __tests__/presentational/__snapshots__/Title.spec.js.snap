// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component should match snapshot when title does not include free label 1`] = `
<div
  className="event__title"
>
  testText
  <span
    className="event__title--free"
  />
</div>
`;

exports[`Event Component should match snapshot when title includes free label 1`] = `
<div
  className="event__title"
>
  testText
  <span
    className="event__title--free"
  >
     FREE!!!
  </span>
</div>
`;
